### ISSUE TITLE
[BUG] worker fixture not being shared between tests when using object value

### ISSUE THREAD

**Issue by annervisser** [microsoft/playwright#21999]
### System info
- Playwright Version: 1.32.1
- Operating System: All (used Ubuntu)
- Browser: All (used chrome)
- Other info:

### Source code

- [x] I provided exact source code that allows reproducing the issue locally.

**Link to the GitHub repository with the repro**

https://github.com/annervisser/playwright-object-fixture-repro (see `tests` directory)

**Description**
When using a value that isn't strictly equal (i.e. it doesn't pass `===`), playwright doesn't see it as being identical and thus creates a new worker instance for every test. This makes it impossible to parameterize tests with complex values.

We're using a worker fixture to spin up an environment with certain settings, and want tests that use the same settings to re-use that environment.

**Steps**
- Use an object value for a fixture to parameterize a test
- Override the same value in two separate tests
- Run both tests (with only 1 worker)

**Expected**

A worker fixture using the configured value should run only once, since only 1 unique set is configured

**Actual**

The worker fixture using the value is run twice, even when running with only 1 worker

Output:
```shell
$ npx playwright test --workers=1

Running 2 tests using 1 worker

  ✓  1 [chromium] › test1.spec.ts:5:5 › test1 (5ms)
Creating new user bob 0 0
  ✓  2 [chromium] › test2.spec.ts:9:5 › test2 (5ms)
Creating new user bob 1 0

  2 passed (445ms)
```

**Comment by yury-s** [comment]
This is working as intended, you can only have one `test.use` per scope (file or test.describe). If you have multiple, the last one will overwrite previous values passed to test.use in the same scope. To overcome this in your scenario use separate `test.describe` scopes instead:

```ts
for (const kw of keywords) {
  test.describe(`Test suite ${kw}`, () => {
          test.use({ keyword: kw });
          test(`test`, async ({ keyword }) => {
              await expect(kw).toBe(keyword);
          });
  });
}
```

Will this work for you?

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.
